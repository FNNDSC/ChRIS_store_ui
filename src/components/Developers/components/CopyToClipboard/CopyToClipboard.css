.copy-to-clipboard-btn {
  /* color */
  background: var(--pf-black-100);

  /* display */
  padding: 5px 9px !important;

  /* position */
  position: absolute;
  top: 7px;
  right: 7px;
}
