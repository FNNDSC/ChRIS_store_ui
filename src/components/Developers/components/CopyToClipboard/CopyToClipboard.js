import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Clipboard from 'react-clipboard.js';
import { Icon, OverlayTrigger, Tooltip } from 'patternfly-react';
import './CopyToClipboard.css';

class CopyToClipboard extends Component {
  constructor(props) {
    super(props);
    this.overlayTrigger = React.createRef();
    this.showTooltip = this.showTooltip.bind(this);
  }

  showTooltip() {
    this.overlayTrigger.current.show();
    setTimeout(() => { this.overlayTrigger.current.hide(); }, 600);
  }

  render() {
    return (
      <OverlayTrigger
        placement="bottom"
        ref={this.overlayTrigger}
        onClick={this.showTooltip}
        overlay={(
          <Tooltip
            id="copiedTooltip"
          >
            <strong>Copied!</strong>
          </Tooltip>
        )}
      >
        <Clipboard
          className="copy-to-clipboard-btn btn btn-large"
          data-clipboard-text={this.props.clipboardText}
        >
          <Icon name="copy" />
        </Clipboard>
      </OverlayTrigger>
    );
  }
}

CopyToClipboard.propTypes = { clipboardText: PropTypes.string.isRequired };

export default CopyToClipboard;
