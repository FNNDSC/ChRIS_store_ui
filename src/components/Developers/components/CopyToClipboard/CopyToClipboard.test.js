import React from 'react';
import { mount, shallow } from 'enzyme';
import Clipboard from 'react-clipboard.js';
import CopyToClipboard from './CopyToClipboard';


describe('CopyToClipboard', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CopyToClipboard clipboardText="" />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Clipboard component', () => {
    expect(wrapper.find(Clipboard)).toHaveLength(1);
  });

  it('should render copy Icon component with name', () => {
    expect(wrapper.find('Icon').prop('name')).toEqual('copy');
  });

  it('should render OverlayTrigger component', () => {
    expect(wrapper.find('OverlayTrigger')).toHaveLength(1);
  });
});

describe('mounted CopyToClipboard', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<CopyToClipboard clipboardText="" />);
  });

  it('renders the data-clipboard-text attribute', () => {
    wrapper.setProps({ clipboardText: 'text' });
    const clipboardElement = wrapper.find(Clipboard);
    expect(clipboardElement.prop('data-clipboard-text')).toEqual('text');
  });

  it('should show tooltip when the copy button is clicked', () => {
    const overlayTrigger = wrapper.find('OverlayTrigger');
    overlayTrigger.simulate('click');
    expect(overlayTrigger.instance().state.show).toEqual(true);
  });
});
