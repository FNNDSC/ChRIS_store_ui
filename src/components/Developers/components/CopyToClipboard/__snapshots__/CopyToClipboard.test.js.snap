// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyToClipboard should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CopyToClipboard
    clipboardText=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ClipboardButton
        className="copy-to-clipboard-btn btn btn-large"
        data-clipboard-text=""
        onClick={[Function]}
>
        <Icon
                name="copy"
                type="fa"
        />
</ClipboardButton>,
      "defaultOverlayShown": false,
      "onClick": [Function],
      "overlay": <Tooltip
        bsClass="tooltip"
        id="copiedTooltip"
        placement="right"
>
        <strong>
                Copied!
        </strong>
</Tooltip>,
      "placement": "bottom",
      "trigger": Array [
        "hover",
        "focus",
      ],
    },
    "ref": Object {
      "current": null,
    },
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Icon
          name="copy"
          type="fa"
/>,
        "className": "copy-to-clipboard-btn btn btn-large",
        "data-clipboard-text": "",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "copy",
          "type": "fa",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ClipboardButton
          className="copy-to-clipboard-btn btn btn-large"
          data-clipboard-text=""
          onClick={[Function]}
>
          <Icon
                    name="copy"
                    type="fa"
          />
</ClipboardButton>,
        "defaultOverlayShown": false,
        "onClick": [Function],
        "overlay": <Tooltip
          bsClass="tooltip"
          id="copiedTooltip"
          placement="right"
>
          <strong>
                    Copied!
          </strong>
</Tooltip>,
        "placement": "bottom",
        "trigger": Array [
          "hover",
          "focus",
        ],
      },
      "ref": Object {
        "current": null,
      },
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Icon
            name="copy"
            type="fa"
/>,
          "className": "copy-to-clipboard-btn btn btn-large",
          "data-clipboard-text": "",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "copy",
            "type": "fa",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
